doc/reference/reference_lua/ulid.rst

```diff
@@ -0,0 +1,523 @@
+..  _ulid-module:
+
+Module ulid
+===========
+
+Since version 3.6.0.
+
+.. _ulid-module-overview:
+
+Overview
+--------
+
+The ``ulid`` module implements ULID (Universally Unique Lexicographically
+Sortable Identifier) support in Tarantool. A ULID is a 128-bit identifier
+consisting of:
+
+* a 48-bit timestamp in milliseconds since the Unix epoch
+* an 80-bit random (entropy) component
+
+ULID strings are encoded using **Crockford Base32**, a compact and
+human-friendly alphabet that excludes visually ambiguous characters.
+
+In binary form, ULIDs are represented as 16-byte values in **big-endian**
+byte order. This ensures that the lexicographical order of binary ULIDs
+matches their chronological order, in accordance with the ULID specification.
+
+ULIDs have several useful properties:
+
+* They are lexicographically sortable by creation time;
+* They fit entirely into 26 ASCII characters;
+* They avoid visually ambiguous symbols (``I``, ``L``, ``O``, ``U``).
+* They have 128 bits of total uniqueness-same as UUID v4.
+
+Tarantool uses a *monotonic* ULID generator. This ensures that multiple ULIDs
+created within the same millisecond are strictly increasing and preserve
+sort order: for any two ULIDs generated in the same millisecond, the one
+created later is greater than the earlier one.
+
+Internally, the monotonic generator keeps the last generated ULID for the
+current millisecond and increments the 80-bit random part for each subsequent
+ULID. A real overflow of the random part can happen only after generating
+``2^80`` ULIDs within the same millisecond, which is practically impossible
+on real hardware. However, for strict correctness of the ULID specification
+and to avoid silent wrap-around, the implementation detects this overflow
+and fails the next generation attempt with a Lua error
+(``ULID random component overflow``).
+
+To use this module, run the following command:
+
+..  code-block:: lua
+
+    ulid = require('ulid')
+
+.. _ulid-module-comparison:
+
+Comparison
+----------
+
+ULID objects support the full set of Lua comparison operators:
+
+* ``==`` and ``~=`` - equality and inequality;
+* ``<`` and ``<=`` - lexicographical comparison;
+* ``>`` and ``>=`` - lexicographical comparison.
+
+The comparison is based on the internal 16-byte representation in
+big-endian order and is consistent with the ULID specification:
+for ULIDs created by the monotonic generator, later ULIDs are greater
+than earlier ones, including ULIDs generated within the same millisecond.
+
+Comparison works both between ULID objects and between a ULID object
+and a ULID string:
+
+* ``u1 == u2`` compares two ULID objects directly;
+* ``u1 == "01..."`` converts the string to ULID and compares values;
+* ``u1 < "01..."`` or ``"01..." < u1`` convert the string argument to ULID
+  and perform lexicographical comparison.
```

Comment:
```suggestion
* ``u1 == u2`` compares two ULID objects directly.
* ``u1 == "01..."`` converts the string to ULID and compares values.
* ``u1 < "01..."`` or ``"01..." < u1`` convert the string argument to ULID
  and perform lexicographical comparison.
```
https://www.tarantool.io/ru/doc/latest/contributing/docs/style/#ending-punctuation-in-lists-and-tables